xscen/ensembles.py

```diff
@@ -59,62 +66,143 @@ def ensemble_stats(
+    statistics_to_compute = list(statistics.keys())
+    if "robustness_categories" in statistics_to_compute:
+        if "robustness_fractions" in statistics_to_compute:
+            statistics_to_compute.remove("robustness_fractions")
+        elif "robustness_fractions" not in statistics["robustness_categories"]:
+            raise ValueError(
+                "'robustness_categories' requires 'robustness_fractions' to be computed. "
+                "Either add 'robustness_fractions' to the statistics dictionary or "
+                "add 'robustness_fractions' under the 'robustness_categories' dictionary."
+            )
+
+    for stat in statistics_to_compute:
+        stats_kwargs = deepcopy(statistics.get(stat) or {})
         logger.info(
-            f"Creating ensemble with {len(datasets)} simulations and calculating {stat}."
+            f"Calculating {stat} from an ensemble of {len(ens.realization)} simulations."
         )
-        if (
-            weights is not None
-            and "weights" in inspect.getfullargspec(getattr(ensembles, stat))[0]
-        ):
-            stats_kwargs["weights"] = weights.reindex_like(ens.realization)
-        if "ref" in stats_kwargs:
-            stats_kwargs["ref"] = ens.loc[stats_kwargs["ref"]]
-
-        if stat == "change_significance":
+
+        # Workaround for robustness_categories
+        real_stat = None
+        if stat == "robustness_categories":
+            real_stat = "robustness_categories"
+            stat = "robustness_fractions"
+            categories_kwargs = deepcopy(stats_kwargs)
+            categories_kwargs.pop("robustness_fractions", None)
+            stats_kwargs = deepcopy(
+                stats_kwargs.get("robustness_fractions", None)
+                or statistics.get("robustness_fractions", {})
+            )
+
+        if weights is not None:
+            if "weights" in inspect.getfullargspec(getattr(ensembles, stat))[0]:
+                stats_kwargs["weights"] = weights.reindex_like(ens.realization)
+            else:
+                warnings.warn(
+                    f"Weighting is not supported for '{stat}'. The results may be incorrect."
+                )
+
+        # FIXME: change_significance is deprecated and will be removed in xclim 0.49.
+        if stat in [
+            "change_significance",
+            "robustness_fractions",
+            "robustness_categories",
+        ]:
+            # FIXME: This can be removed once change_significance is removed.
+            #  It's here because the 'ref' default was removed for change_significance in xclim 0.47.
+            stats_kwargs.setdefault("ref", None)
+            if (stats_kwargs.get("ref") is not None) and len(statistics.keys()) > 1:
+                raise ValueError(
+                    f"The input requirements for '{stat}' when 'ref' is specified are not compatible with other statistics."
```

Comment:
so it's okay to put ref, but only if we only request one stat ?

What if we want `robustness_categories`, then we have to pass `'robustness_fractions'`, which means we can't use ref. Is that right ?

maybe a bit more guiding on ref would help in the docstring.

Reply:
My error, it should have been `len(statistics_to_compute)`. Basically if you want to use `ref`, then you can't call other statistics such as `ensemble_percentiles`.

I've also added more details on `ref`.